File: backend/person/services/person_service.py
from ..tasks import PersonTask
from ..models import Person

class PersonService:
    @staticmethod
    def create(dto):
        return PersonTask.create(dto)
    
    @staticmethod
    def update(person_id, dto):
        return PersonTask.update(person_id, dto)
    
    @staticmethod
    def delete(person_id):
        return PersonTask.delete(person_id)
    
    @staticmethod
    def search(filters):
        return PersonTask.search(filters)
    
    @staticmethod
    def get_ideal_weight(person: Person) -> float:
        return PersonTask.calculate_ideal_weight(person.altura, person.sexo)

File: frontend/src/app/services/person.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Person } from '../models/person.model';

@Injectable({
  providedIn: 'root'
})
export class PersonService {

  private apiUrl = 'http://localhost:8000/api/pessoas/';

  constructor(private http: HttpClient) {}

  list(): Observable<Person[]> {
    return this.http.get<Person[]>(this.apiUrl);
  }

  create(person: Person): Observable<Person> {
    return this.http.post<Person>(this.apiUrl, person);
  }

  update(id: number, person: Person): Observable<Person> {
    return this.http.put<Person>(`${this.apiUrl}${id}/`, person);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}${id}/`);
  }

  getIdealWeight(id: number): Observable<{ peso_ideal: number }> {
    return this.http.get<{ peso_ideal: number }>(
      `${this.apiUrl}${id}/calculo-peso/`
    );
  }

  getById(id: number): Observable<Person> {
    return this.http.get<Person>(`${this.apiUrl}${id}/`);
  }
}
